app/controllers/users/registrations_controller.rb

```diff
@@ -0,0 +1,41 @@
+class Users::RegistrationsController < Devise::RegistrationsController
+  skip_before_action :verify_authenticity_token
+  
+  def new
+    build_resource({})
+    resource.build_profile
+    respond_with resource
+  end
+
+  def create
+    build_resource(sign_up_params)
+
+    if resource.valid? && resource.save
+      if resource.role == 'customer'
+        sign_up(resource_name, resource)
+        set_flash_message! :notice, :signed_up
+        redirect_to new_profile_path
+      else
```

Comment:
It looks like the logic inside the if & the else is the same, the only thing that differs is the redirect path. So what you could do is:
```ruby
sign_up(resource_name, resource)
set_flash_message! :notice, :signed_up

redirect_path = rescource.customer? ? new_profile_path : root_path
redirect_to redirect_path
```

This uses an inline if that looks like: `boolean ? behaviour_if_boolean_is_true : behaviour_if_boolean_is_false`